app/domain/authentication/authn_jwt/jwt_fetch_jwks_uri_signing_key.rb

```diff
@@ -0,0 +1,51 @@
+require 'uri'
+require 'net/http'
+
+module Authentication
+  module AuthnJwt
+
+    JwtFetchJwksUriSigningKey = CommandClass.new(
+      dependencies: {
+        logger: Rails.logger,
+        discover_identity_provider: OAuth::DiscoverIdentityProvider.new,
+        fetch_secrets: Conjur::FetchRequiredSecrets.new
+      },
+      inputs: %i[authenticator_input]
+    ) do
+      def call
+        jwks_uri
+        fetch_jwks_keys
+      end
+
+      private
+
+      def jwks_uri
+        @logger.debug(LogMessages::Authentication::AuthnJwt::JwksUriResourceNameConfiguration.new(resource_id))
+        @jwks_uri_secret ||= @fetch_secrets.(resource_ids: [resource_id])
+        @jwks_uri ||=@jwks_uri_secret[resource_id]
+      end
+
+      def resource_id
+        "#{@authenticator_input.account}:variable:conjur/#{@authenticator_input.authenticator_name}/#{@authenticator_input.service_id}/#{JWKS_URI_RESOURCE_NAME}"
+      end
+
+      def fetch_jwks_keys
+        uri = URI(@jwks_uri)
+        @logger.debug(LogMessages::Authentication::AuthnJwt::FetchingJwksFromJwksUri.new)
+        response = Net::HTTP.get_response(uri)
+        jwks = {
+          keys: JSON.parse(response.body)['keys']
+        }
+        # TODO: this field should be set on ValidateAndDecode by the algorithm of the JWT itself
+        algs = ["RS256"]
```

Comment:
we have a very similar code block in `FetchProviderKeys`. can we use it instead?

Reply:
Do you mean I should send the `jwks` and `algs` as params to `FetchProviderKeys`?
I can live with that

Reply:
i was actually saying that you should use `FetchProviderKeys` instead of implementing the logic be yourself. `FetchProviderKeys` receives a provider_uri as input and returns a `ProviderKeys` object that contains the jwks and algs. isn't this exactly what you are doing here?

Reply:
Actually thats not what im doing here, in the case of `jwks-uri` we support vendors that have unique `jwks-uri` that is unrelated to the "well-known" oidc uri (which is referred in this pr as "provider-uri). thats why `FetchJwksUriSigningKey` has no use of the oidc third party

Reply:
still. looks like a lot of the logic can be centralized. if needed, let's sit together and see how it can be done